.clang-tidy

```diff
@@ -22,4 +22,6 @@ CheckOptions:
     value:           true
   - key:             misc-non-private-member-variables-in-classes.IgnoreClassesWithAllMemberVariablesBeingPublic
     value:           true
+  - key:             readability-else-after-return.WarnOnConditionVariables
```

Comment:
The docs say:

> When true, the check will attempt to refactor a variable defined inside the condition of the if statement that is used in the else branch defining them just before the if statement. This can only be done if the if statement is the last statement in its parents scope. Default value is true.

Can you give an example in terms of code?

Reply:
```
{
  if (x)
    return y;
  else
    z;
}
```

must be written as this if the rule is enabled:

```
{
  if (x)
    return y;
  z;
}
```

This is even the case is `z` is dependent on `x`. The rule is just overzealous and doesn't match our coding style.